pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -478,7 +478,7 @@ func (facts *PipelineRunFacts) GetPipelineTaskStatus() map[string]string {
 			case t.isSuccessful():
 				s = v1.TaskRunReasonSuccessful.String()
 			// execution status is Failed when a task has succeeded condition with status set to false
-			case t.isConditionStatusFalse():
+			case t.haveAnyRunsFailed():
 				s = v1.TaskRunReasonFailed.String()
```

Comment:
It's confusing seeing this reused for both `TaskRuns` and `CustomRuns`, I think we should create generic `RunReason...` that are reused by all runtypes - this can be done in future work though

```suggestion
				s = v1.RunReasonFailed.String()
```

Reply:
agreed, but I think this belongs in a separate PR